src/frontend/planner_test/tests/testdata/output/cast.yaml

```diff
@@ -80,3 +80,35 @@
     └─BatchExchange { order: [], dist: Single }
       └─BatchSimpleAgg { aggs: [count] }
         └─BatchScan { table: t, columns: [], distribution: SomeShard }
+- name: composite type cast error message (array)
+  sql: |
+    select '[]'::int[]::bytea[];
+  binder_error: |
+    Failed to bind expression: CAST(CAST('[]' AS INT[]) AS BYTEA[])
+
+    Caused by these errors (recent errors listed first):
+      1: cannot cast type "integer[]" to "bytea[]" in Explicit context
+      2: cannot cast type "integer" to "bytea" in Explicit context
+- name: composite type cast error message (struct)
+  sql: |
+    create table t (v struct<a struct<b int>, c bool>);
+    select v::struct<d struct<e bytea>, f bool> from t;
+  binder_error: |
+    Failed to bind expression: CAST(v AS STRUCT<d STRUCT<e BYTEA>, f BOOLEAN>)
+
+    Caused by these errors (recent errors listed first):
+      1: cannot cast type "struct<a struct<b integer>, c boolean>" to "struct<d struct<e bytea>, f boolean>" in Explicit context
+      2: cannot cast type "struct<b integer>" to "struct<e bytea>" in Explicit context
+      3: cannot cast type "integer" to "bytea" in Explicit context
+- name: composite type cast error message (map)
+  sql: |
+    create table t (v map(int, int));
+    select v::map(int, bytea) from t;
+  binder_error: |
+    Failed to bind expression: CAST(v AS MAP(INT,BYTEA))
+
+    Caused by these errors (recent errors listed first):
+      1: cannot cast type "map(integer,integer)" to "map(integer,bytea)" in Explicit context
+      2: cannot cast type "struct<key integer, value integer>[]" to "struct<key integer, value bytea>[]" in Explicit context
+      3: cannot cast type "struct<key integer, value integer>" to "struct<key integer, value bytea>" in Explicit context
+      4: cannot cast type "integer" to "bytea" in Explicit context
```

Comment:
This might be a little confusing to users 🤪

Reply:
Updated. Check check again.